backend/plugin/db/trino/sync.go

```diff
@@ -2,96 +2,557 @@ package trino
 
 import (
 	"context"
+	"database/sql"
 	"fmt"
+	"log/slog"
+	"strings"
+	"time"
 
 	"github.com/pkg/errors"
 
+	"github.com/bytebase/bytebase/backend/common/log"
 	"github.com/bytebase/bytebase/backend/plugin/db"
 
 	storepb "github.com/bytebase/bytebase/proto/generated-go/store"
 )
 
-func (d *Driver) SyncInstance(ctx context.Context) (*db.InstanceMetadata, error) {
-	version, err := d.getVersion(ctx)
+type contextKey string
+
+const (
+	explicitDBNameKey contextKey = "explicitDbName"
+	databaseKey       contextKey = "database"
+	resourceIDKey     contextKey = "resourceID"
+	nameKey           contextKey = "name"
+	requestNameKey    contextKey = "requestName"
+	databaseNameKey   contextKey = "databaseName"
+	checkKey          contextKey = "check"
+	pathKey           contextKey = "path"
+	statementKey      contextKey = "statement"
+	resourceIDKey2    contextKey = "resourceId"
+)
+
+func (d *Driver) processRows(ctx context.Context, query string, handler func(rows *sql.Rows) error) error {
+	queryCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	rows, err := d.db.QueryContext(queryCtx, query)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get version")
+		return err
+	}
+
+	defer rows.Close()
+	if err := rows.Err(); err != nil {
+		return err
+	}
+	return handler(rows)
+}
+
+func scanStringValues(rows *sql.Rows, logErrorPrefix string) []string {
+	var results []string
+	for rows.Next() {
+		var value string
+		if err := rows.Scan(&value); err == nil && value != "" {
+			results = append(results, value)
+		}
 	}
 
-	var databases []*storepb.DatabaseSchemaMetadata
+	if err := rows.Err(); err != nil {
+		slog.Warn(logErrorPrefix, log.BBError(err))
```

Comment:
return error if encountering error like what we do in other database engines syncing